trio/_channel.py

```diff
@@ -109,28 +138,30 @@ def statistics(self):
 
 
 @attr.s(eq=False, repr=False)
-class MemorySendChannel(SendChannel, metaclass=NoPublicConstructor):
-    _state = attr.ib()
-    _closed = attr.ib(default=False)
+class MemorySendChannel(
+    SendChannel[SendType], Generic[SendType], metaclass=NoPublicConstructor
```

Comment:
I don't think the `Generic` here is necessary, FWIW (same for `MemoryReceiveChannel`)

Reply:
ah of course, thanks!